eventsources/eventing.go

```diff
@@ -422,6 +445,23 @@ func (e *EventSourceAdaptor) run(ctx context.Context, servers map[apicommon.Even
 						if err != nil {
 							return err
 						}
+
+						if filter, ok := filters[getFilterKey(s.GetEventSourceType(), s.GetEventName())]; ok {
+							dataMap := make(map[string]interface{})
+							err := json.Unmarshal(data, &dataMap)
+							if err != nil {
+								return err
```

Comment:
If err is no nil (the data format might not be json in most of case), don't return err, give a warning instead, and skip dispatching the message.